lib/rules/max-statements.js

```diff
@@ -84,9 +84,18 @@ module.exports = {
          */
         function reportIfTooManyStatements(node, count, max) {
             if (count > max) {
+                const messageEnd = " has too many statements ({{count}}). Maximum allowed is {{max}}.";
+                let name = "This function";
+
+                if (node.id) {
+                    name = `Function '${node.id.name}'`;
+                } else if (node.parent.type === "MethodDefinition" || node.parent.type === "Property") {
+                    name = `Function '${node.parent.key.name}'`;
+                }
```

Comment:
@nzakas `({["foo"]: function() { ... }})`; in this case, `node.parent.key` does not have `name` property since it's a Literal node. So `Function '${node.parent.key.name}'` should become `Function undefined`.

----

By the way, I made a utility function for this in #7435. What about the `getFunctionNameWithKind(node)` function?